pkg/pillar/base/execwrapper.go

```diff
@@ -0,0 +1,150 @@
+package base
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime/debug"
+	"strings"
+	"time"
+)
+
+const (
+	// Will wait for 1000s max for a call instead of waiting forever
+	maxTimeOutLimit = 1000
+	timeOutLimit    = 100
+)
+
+//ExecCommand executes the given command. Times out if timeOut = true.
+// Also if the exec is taking long, it will update the caller agent's touch file.
+func ExecCommand(log *LogObject, combined bool, timeOut bool, command string, arg ...string) ([]byte, error) {
```

Comment:
This may be a somewhat controversial opinion, but I really find it difficult to remember what these 2nd and 3d argument stand for. So... would it make sense to have syntactic sugar on top called ExecCommandForever(), etc.

And for the combined use the .CombinedOutput() approach?

WDYT @eriknordmark ?

Reply:
Done.